cgi-bin/DW/Controller/Support/History.pm

```diff
@@ -0,0 +1,175 @@
+            foreach my $row ( @$reqlist ) {
+                $reqs{$row->[0]} = {
+                    spid => $row->[0],
+                    winner => $points{$row->[0]}->[0],
+                    points => $points{$row->[0]}->[1]+0,
+                    subject => LJ::ehtml($row->[1]),
+                    state => $row->[2],
+                    spcatid => $row->[3],
+                    requserid => $row->[4],
+                    timecreate => $row->[5],
+                    timetouched => $row->[6],
+                    timelasthelp => $row->[7],
+                    reqemail => LJ::ehtml( $row->[8] ),
+                };
+                push @userids, $row->[4] if $row->[4];
+            }
+            my $us = @userids ? LJ::load_userids( @userids ) : undef;
+
+            # get categories
+            my $cats = LJ::Support::load_cats();
+
+            $ret .= '<table><thead><tr><th>Summary</th><th>State</th><th>Answered By</th>' .
+                    '<th>Category</th><th>Opened By</th><th>Time Opened</th></tr></thead>';
+            foreach my $id ( sort { $a <=> $b } keys %reqs ) {
+                # verify user can see this category (public_read or has supportread in it)
+                next unless $cats->{$reqs{$id}->{spcatid}}{public_read} ||
+                    LJ::Support::can_read_cat($cats->{$reqs{$id}->{spcatid}}, $remote);
+                my $status = $reqs{$id}->{state} eq "closed" ? "closed" :
+                             LJ::Support::open_request_status($reqs{$id}->{timetouched},
+                                                              $reqs{$id}->{timelasthelp});
+                my $answered = 0;
+                my $points = 0;
+                my $answeredby = '-';
+                if ($reqs{$id}->{state} eq 'closed' && $reqs{$id}->{winner}) {
+                    $answeredby = LJ::ljuser( $reqs{$id}->{winner} );
+                    $points = $reqs{$id}->{points};
+                    $answered = 1;
+                }
+                $ret .= $reqs{$id}->{requserid} ? LJ::ljuser($us->{$reqs{$id}->{requserid}}) : $reqs{$id}->{reqemail};
+                $ret .= "</td>\n<td align='center'>";
+                $ret .= LJ::mysql_time($reqs{$id}->{timecreate});
+                push @{ $vars->{reqs} }, {
+                    id => $id,
+                    spid => $reqs{$id}->{spid},
+                    subject => $reqs{$id}->{subject},
+                    status => $status,
+                    answeredby => $answeredby,
+                    points => $points,
+                    answered => $answered,
+                    catname => $cats->{$reqs{$id}->{spcatid}}{catname},
+                    openedby => $reqs{$id}->{requserid} ? LJ::ljuser($us->{$reqs{$id}->{requserid}}) : $reqs{$id}->{reqemail},
+                    timeopened => LJ::mysql_time( $reqs{$id}->{timecreate} )
+                };
+            }
+        } else {
+            $vars->{noresults} = 1;
+        }
+    } else {
```

Comment:
This else block should only be executed if $fullsearch is false, instead of just the final statement.  e.g.:
```
} elsif ( ! $fullsearch ) { ... }
```
(I realize this is sacrificing a bit of readability by changing an 'unless' to an 'if !', but it's the easiest way to do it, I think, given that Perl doesn't have an 'elsunless' ;)